### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by AndreySinelnikov**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw3/AbstractExercise.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.io.IOException;
+import java.util.Properties;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+
+public abstract class AbstractExercise {
+    protected WebDriver webdriver;
```
Comment (DmitryKhodakovsky) [inline src/test/java/com/epam/tc/hw3/AbstractExercise.java]: `webDriver` || `driver` :)

File: src/test/java/com/epam/tc/hw3/AbstractExercise.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    protected SoftAssertions softly;
+    public Properties props;
```
Comment (DmitryKhodakovsky) [inline src/test/java/com/epam/tc/hw3/AbstractExercise.java]: Why this variable is public?

File: src/test/java/com/epam/tc/hw3/pages/BasePage.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public abstract class BasePage {
+    protected WebDriver webdriver;
+    protected UpperNavPanel upperNavPanel;
+    protected LeftNavPanel leftNavPanel;
+    protected String baseURL;
+
+    public BasePage(WebDriver webdriver, String url) {
```
Comment (DmitryKhodakovsky) [inline src/test/java/com/epam/tc/hw3/pages/BasePage.java]: `BasePage` could be extended from `BaseComponent`

File: src/test/java/com/epam/tc/hw3/pages/BasePage.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        PageFactory.initElements(webdriver, this);  // allows FindBy for page elements
+        upperNavPanel = new UpperNavPanel(webdriver);
+        leftNavPanel = new LeftNavPanel(webdriver);
+        this.webdriver = webdriver;
+        baseURL = url;
+    }
+
+    public void load() {
```
Comment (DmitryKhodakovsky) [inline src/test/java/com/epam/tc/hw3/pages/BasePage.java]: `open`

**Review by DmitryKhodakovsky — APPROVED** [review]
Approved.
Some minor comments left. 

Please merge branch to the `main`

File: src/test/java/com/epam/tc/hw3/AbstractExercise.java (hunk 1 above)
Comment (AndreySinelnikov) [reply src/test/java/com/epam/tc/hw3/AbstractExercise.java]: renamed to webDriver in all files

File: src/test/java/com/epam/tc/hw3/AbstractExercise.java (hunk 2 above)
Comment (AndreySinelnikov) [reply src/test/java/com/epam/tc/hw3/AbstractExercise.java]: forgot to revert it to protected after experimentation, fixed

File: src/test/java/com/epam/tc/hw3/pages/BasePage.java (hunk 4 above)
Comment (AndreySinelnikov) [reply src/test/java/com/epam/tc/hw3/pages/BasePage.java]: renamed

File: src/test/java/com/epam/tc/hw3/pages/BasePage.java (hunk 3 above)
Comment (AndreySinelnikov) [reply src/test/java/com/epam/tc/hw3/pages/BasePage.java]: Made both classes abstract with protected constructors, AbstractPage extends AbstractComponent now

**Pull request merged**